pkg/resource/table/hooks.go

```diff
@@ -140,139 +140,164 @@ func (rm *resourceManager) customUpdateTable(
 	ko := desired.ko.DeepCopy()
 	rm.setStatusDefaults(ko)
 
+	// Quoting from https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateTable.html
+	//
+	// You can only perform one of the following operations at once:
+	// - Modify the provisioned throughput settings of the table.
+	// - Enable or disable DynamoDB Streams on the table.
+	// - Remove a global secondary index from the table.
+	// - Create a new global secondary index on the table. After the index begins
+	//   backfilling, you can use UpdateTable to perform other operations.
+	//
+	// UpdateTable is an asynchronous operation; while it is executing, the table
+	// status changes from ACTIVE to UPDATING. While it is UPDATING, you cannot
+	// issue another UpdateTable request. When the table returns to the ACTIVE state,
+	// the UpdateTable operation is complete.
```

Comment:
Ah this is the same as EKS clusters. For that, I threw [a requeue error](https://github.com/aws-controllers-k8s/eks-controller/blob/main/pkg/resource/cluster/hook.go#L59) after making an update, giving the system a 15 second time to wait for the EKS service to register the update, start the async update and update the cluster status description. Do we need to do the same thing here, or does the Dynamo table immediately start updating and update the status? That is, if I describe the table instantly after calling `UpdateTable` will it show as `UPDATING`?

Reply:
Gotcha! Dynamodb dosen't always instantly start updating, i added the same code path you have with EKS.